src/main/scala/org/apache/spark/sql/pulsar/PulsarHelper.scala

```diff
@@ -55,6 +62,12 @@ private[pulsar] case class PulsarHelper(
   private var topics: Seq[String] = _
   private var topicPartitions: Seq[String] = _
 
+  // We can do this because pulsarAdmin will only be called if latestOffset is called
+  // and there should be an exception thrown in PulsarProvider if maxBytes is set,
+  // and maxBytes is not set
+  private lazy val pulsarAdmin = PulsarAdmin.builder().serviceHttpUrl(adminUrl.get).build()
```

Comment:
pulsarAdmin is only used in admissionControlHelper, does it make more sense to make it a member of PulsarAdmissionControlHelper?

Reply:
I think this makes sense, will refactor